### PR TITLE
[Feature] #33 EditorDetailView Data Binding

### LINKED ISSUE #33: [Feature] EditorDetailView Data Binding
# 유저는 에디터 콘텐츠를 확인할 수 있다
- 에디터 콘텐츠 data를 불러온다

## 🤔 EditorDetailView Data Binding

- [ ] make EditorDetailViewModel
- [ ] binding EditorDetailView Data

## 👀 Related issues

- Related issue1 (#26)

### PR DESCRIPTION
**Pull request by rriver2**
# Issue Number
🔒 Close #33

## Changes

- Changed files
    - campair/campair/View/Editor/EditorDetailView.swift
    - campair/campair/ViewModel/EditorViewModel.swift

## New features

- make EditorDetailViewModel
- binding EditorDetailView Data

## Review points

- EditorDetailViewModel class structure
- EditorDetailViewModel data binding
image
image

## Questions

- @sustainable-git 
I think we should change swiftLint that "Line should be 200 characters or less". Because I am experiencing inconvenience when working with fake data.
스크린샷 2022-06-14 오후 10 30 58

## Checklist

- [x] Is the branch you are merging on correct?
- [x] Do you comply with coding conventions?
- [x] Are there any changes not related to PR?
- [x] Has my code been self-reviewed?

### CODE REVIEW

**Comment by unnnyong** [comment]
스크린샷이 있으니 어느 화면의 수정인지 확인할 수 있어서 너무 좋네요 !!!!!!! 😍 😍 😍 😍

**Comment by unnnyong** [comment]
PR diff에서 파일 변경 목록은 확인할 수 있으니까, 본문에서의 Changed files 항목은 생략하면 어떨까 ,,,,하는 의견을 남겨봅니다 : )

File: campair/campair/View/Editor/EditorDetailView.swift [hunk 1]
```diff
@@ -8,14 +8,17 @@
...
             VStack(alignment: .leading, spacing: 0) {
                 VStack(alignment: .leading, spacing: 0) {
-                    ForEach(0..<2) {index in
-                        EditorRecommendEquipmentView()
-                        if(index != 1){
+                    let equipmentContents = self.editorDetailContent.content.equipmentContents
+                    ForEach(equipmentContents.indices) {index in
+                        let equipmentContent = equipmentContents[index]
+                        EditorRecommendEquipmentView(equipmentContent: equipmentContent)
+                        if(index != equipmentContents.count - 1) {
```
Comment (unnnyong) [inline campair/campair/View/Editor/EditorDetailView.swift:21]:
```suggestion
                        if (index != equipmentContents.count - 1) {
```
정말로 사소~

File: campair/campair/View/Editor/EditorDetailView.swift [hunk 2]
```diff
@@ -8,14 +8,17 @@
...
-            OpeningEditorView()
+            OpeningEditorView(openingSection: self.editorDetailContent.content.opening)
             VStack(alignment: .leading, spacing: 0) {
                 VStack(alignment: .leading, spacing: 0) {
-                    ForEach(0..<2) {index in
-                        EditorRecommendEquipmentView()
-                        if(index != 1){
+                    let equipmentContents = self.editorDetailContent.content.equipmentContents
+                    ForEach(equipmentContents.indices) {index in
+                        let equipmentContent = equipmentContents[index]
```
Comment (unnnyong) [inline campair/campair/View/Editor/EditorDetailView.swift:19]: out of range가 우려됩니다 ~
조건문으로 index확인을 추천드려요

File: campair/campair/View/Editor/EditorDetailView.swift [hunk 3]
```diff
@@ -31,18 +34,19 @@ struct EditorDetailView: View {
 }
 
 struct OpeningEditorView: View {
+    let openingSection : OpeningSection
```
Comment (unnnyong) [inline campair/campair/View/Editor/EditorDetailView.swift:37]:
```suggestion
    private let openingSection : OpeningSection
```

🔒

File: campair/campair/View/Editor/EditorDetailView.swift (hunk 2 above)
Comment (sustainable-git) [reply campair/campair/View/Editor/EditorDetailView.swift:19]: If it is not a mock code, please check the length of array.
```swift
guard equipmentContents.count > index
```

File: campair/campair/View/Editor/EditorDetailView.swift [hunk 4]
```diff
@@ -31,18 +34,19 @@ struct EditorDetailView: View {
...
-                    Text("차크닉 시작하기")
+                    Text("\(self.openingSection.cardTitle)")
                         .font(.system(size: 15))
                         .bold()
                         .foregroundColor(Color.customPaintingTextPink)
-                    Text("차로 떠나는 피크닉")
+                    Text("\(self.openingSection.cardSubTitle)")
                         .foregroundColor(Color.white)
                         .font(.system(size: 28))
                     // heavy -> extraBold : 800 숫자로 맞춰야 하는데 숫자를 fontWeight으로 주는 방법을 모르겠습니다.
```
Comment (sustainable-git) [inline campair/campair/View/Editor/EditorDetailView.swift:52]: Please erase comment if it is resolved

File: campair/campair/View/Editor/EditorDetailView.swift (hunk 1 above)
Comment (rriver2) [reply campair/campair/View/Editor/EditorDetailView.swift:21]: thank you ~

File: campair/campair/ViewModel/EditorViewModel.swift [hunk 5]
```diff
@@ -8,13 +8,25 @@
...
 final class EditorViewModel: ObservableObject {
     private let editorUseCase = EditorUseCase(repository: EditorRepository())
     @Published var titleText = "test"
+    
     func testButtonTouched() {
         self.editorUseCase.getEditorCollection { [weak self] editorCollection in
+            
+            print(editorCollection)
             // test code
             self!.titleText = editorCollection.editorContents[0].equipmentContents[2].recommendedEquipments[0].content
```
Comment (sustainable-git) [inline campair/campair/ViewModel/EditorViewModel.swift:21]: Don't use `self!`
You should use optional binding here
```swift
guard let self = self else { return }
```

**Review by sustainable-git — CHANGES_REQUESTED** [review]

File: campair/campair/View/Editor/EditorDetailView.swift (hunk 2 above)
Comment (rriver2) [reply campair/campair/View/Editor/EditorDetailView.swift:19]: The green code area is the code I just loaded, but I don't know why it's out of range when I use index with foreach. @unnnyong

File: campair/campair/View/Editor/EditorDetailView.swift (hunk 3 above)
Comment (rriver2) [reply campair/campair/View/Editor/EditorDetailView.swift:37]: 스크린샷 2022-06-15 오전 10 47 48
I wrote down private first time, but I met this error. So I deleted private...

**Review by Gobans — APPROVED** [review]
LGTM! 👏

File: campair/campair/View/Editor/EditorDetailView.swift [hunk 6]
```diff
@@ -120,36 +152,85 @@ struct ShareRecommendedEquipmentView: View {
...
-                .foregroundColor(Color.customSelectedHeartRed)
-        }
-        .padding(.vertical, 20)
-        .background(Color.customCardGray)
-        .cornerRadius(20)
-        .padding(.top, 30)
-    }
-}
-
 struct EditorDetailView_Previews: PreviewProvider {
```
Comment (sustainable-git) [inline campair/campair/View/Editor/EditorDetailView.swift:155]: It's awesome!

**Review by sustainable-git — APPROVED** [review]
GGOODD!

**Pull request merged**